src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -357,9 +358,60 @@ public static string GetDatabaseChangeVector(Transaction tx)
             return Encodings.Utf8.GetString(val.Reader.Base, val.Reader.Length);
         }
 
+        private List<string> GetUnusedList()
+        {
+            using (DocumentDatabase.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            using (var blittableDatabase = DocumentDatabase.ServerStore.Cluster.ReadRawDatabase(context, _name, out _))
+            {
+                if (blittableDatabase.TryGet(nameof(DatabaseRecord.UnusedDatabaseIds), out BlittableJsonReaderArray unused) == false)
+                    return null;
+
+                if (unused == null || unused.Length == 0)
+                    return null;
+                var list = new List<string>();
+                foreach (var item in unused)
+                {
+                    list.Add(item.ToString());
+                }
+
+                return list;
+            }
+        }
+
+        public bool TryRemoveUnusedIds(ref string changeVector)
+        {
+            if (string.IsNullOrEmpty(changeVector))
+                return false;
+
+            var list = GetUnusedList();
```

Comment:
This seems to be really expensive opening a read transaction each time we want to update a change vector.
Maybe we can hold an in-memory list that we will modify in a volatile manner if we need to change it.
The persistent state will be written and read on startup.